service/history/api/describeworkflow/api.go

```diff
@@ -155,6 +155,15 @@ func Invoke(
 			FirstRunId:                   executionInfo.FirstExecutionRunId,
 			VersioningInfo:               executionInfo.VersioningInfo,
 		},
+		WorkflowExtendedInfo: &workflowpb.WorkflowExecutionExtendedInfo{
+			ExecutionExpirationTime: executionInfo.WorkflowExecutionExpirationTime,
+			RunExpirationTime:       executionInfo.WorkflowRunExpirationTime,
+			CancelRequested:         executionInfo.CancelRequested,
+		},
+	}
+
+	if executionInfo.OriginalExecutionRunId != executionState.RunId {
```

Comment:
OriginalExecutionRunId is a new field and old workflows don't have it, so also need to check if the field is non-empty.

Reply:
This is the only way (I was able to find) to find out if workflow was reseted.  There is also `WasReset`, but it is set for "original" workflow, not for the new instance after reset.

Thus, if this is old workflow, this field will be empty. 
In this case this check will not pass (RunId is not empty), and LastResetTime will not be set.
Which is exactly what I would expect, extra check is not needed.

Reply:
for an old workflow OriginalExecutionRunId is empty, RunID is not empty. empty != non-empty, so the check will pass and the LastResetTime will be set right?

Reply:
@gow addedthe IsResetRun() method on mutable state and can be used directly.

Reply:
will use that funciton